server/elasticsearch/src/main/java/org/hawkular/apm/server/elasticsearch/SpanServiceElasticsearch.java

```diff
@@ -192,6 +203,85 @@ public void clear(String tenantId) {
         }
     }
 
+    @Override
+    public Trace getTraceFragment(String tenantId, String id) {
+        Span span = getSpan(tenantId, id);
+
+        if (span == null) {
+            return null;
+        }
+
+        InteractionNode interactionNode = spanToNode(span);
+        interactionNode.setNodes(recursiveTraceFragment(tenantId, span));
+
+        Trace trace = spanToTrace(span);
+        trace.getNodes().add(interactionNode);
+
+        return trace;
+    }
+
+
+    private List<Node> recursiveTraceFragment(String tenantId, Span parent) {
+        List<Span> spanChildren = getChildren(tenantId, parent.getId());
+
+        if (spanChildren == null) {
+            return Collections.emptyList();
+        }
+
+        List<Node> nodes = new ArrayList<>();
+
+        for (Span child: spanChildren) {
+            InteractionNode node = spanToNode(child);
+
+            if (!(parent.serverSpan() && node.getType() == NodeType.Consumer ||
```

Comment:
No, I think the logic needs to go into the getChildren methods to not return server spans - as there are other situations where the getChildren method will be used. So revert this back to just adding the node.